source/Octopus.Server.Client/OctopusServerEndpoint.cs

```diff
@@ -8,9 +8,31 @@ namespace Octopus.Client
     /// </summary>
     public class OctopusServerEndpoint
     {
-        readonly ILinkResolver octopusServer;
-        readonly string apiKey;
-        readonly ICredentials credentials;
+        /// <summary>
+        /// Create an instance with a Token to authenticate.
+        /// </summary>
+        public static OctopusServerEndpoint CreateWithToken(string octopusServerAddress, string bearerToken, ICredentials credentials = null)
```

Comment:
```suggestion
        public static OctopusServerEndpoint CreateWithBearerToken(string octopusServerAddress, string bearerToken, ICredentials credentials = null)
```
and also the assent tests